lib/data_hygiene/duplicate_content_item/version_for_locale.rb

```diff
@@ -0,0 +1,68 @@
+module DataHygiene
+  module DuplicateContentItem
+    class VersionForLocale
+      include DuplicateContentItem::ResultsHelper
+
+      def has_duplicates?
+        number_of_duplicates > 0
+      end
+
+      def number_of_duplicates
+        results[:number_of_duplicates]
+      end
+
+      def results
+        @results ||= build_results
+      end
+
+      def log
+        return unless has_duplicates?
+        message = "#{results[:distinct_content_item_ids]} content items with version for locale conflicts"
+        Airbrake.notify_or_ignore(
+          DuplicateContentItem::DuplicateVersionForLocaleError.new(message),
+          parameters: results
+        )
+      end
+
+    private
+
+      def build_results
+        query_results = ActiveRecord::Base.connection.execute(sql)
+        duplicates = query_results.map do |row|
+          row.symbolize_keys.merge(
+            content_items: content_items_string_to_hash(row["content_items"])
+          )
+        end
+        content_ids = content_ids_from_duplicates(duplicates)
+        content_item_ids = content_item_ids_from_duplicates(duplicates)
+        {
+          distinct_content_items: content_ids.count,
+          content_ids: content_ids,
+          distinct_content_item_ids: content_item_ids.count,
+          content_item_ids: content_item_ids,
+          number_of_duplicates: duplicates.count,
+          duplicates: duplicates
+        }
+      end
+
+      def sql
+        <<-SQL
+          SELECT content_items.content_id,
+            translations.locale,
+            user_facing_versions.number as user_facing_version,
+            ARRAY_AGG(
+              ROW(content_items.id, content_items.updated_at)
+              ORDER BY content_items.updated_at DESC
+            ) as content_items
+          FROM content_items
+          INNER JOIN translations
+            ON translations.content_item_id = content_items.id
+          INNER JOIN user_facing_versions
+            ON user_facing_versions.content_item_id = content_items.id
+          GROUP BY content_id, locale, user_facing_version
+          HAVING COUNT(*) > 1
```

Comment:
I was having trouble visualising what this query was picking up in the database, so I wrote another query which pulls out the (user_facing_version, state) history grouped by content_id where there is a (user_facing_version, locale) pair which is in the history twice.

Also, I tweaked the above query to group by content_id, and it gave me the same number of rows locally :)
```
SELECT content_items.content_id, ARRAY_AGG(DISTINCT content_items.publishing_app) AS publishing_app,
       ARRAY_AGG((user_facing_versions.number, states.name) ORDER BY user_facing_versions.number) AS version_and_state_details
  FROM content_items
 INNER
  JOIN user_facing_versions
    ON content_items.id                                                   = user_facing_versions.content_item_id
 INNER
  JOIN states
    ON content_items.id                                                   = states.content_item_id
 INNER
  JOIN translations
    ON content_items.id                                                   = translations.content_item_id
 GROUP BY content_items.content_id
HAVING COUNT(DISTINCT (user_facing_versions.number, translations.locale)) < COUNT((user_facing_versions.number, translations.locale));
```